ppx_view/test/dune

```diff
@@ -1,6 +1,5 @@
 (library
  (name ppx_view_test)
- (public_name ppx.ppx_view._test_internal)
```

Comment:
Isn’t this unattached a test from a pack?

Reply:
Adding a public name make it being built by `dune build @install` which triggered a build failure in CI because the test dependencies weren't met.

I feel like the whole "inline tests as a testing framework" thing doesn't work very well with the current tooling. Making it a public library just to attach it to a package feels like a hack and has unwanted side effects such as the one I just described.

I have to admit that I'm not sure what the benefit of inline tests is compared to a regular testing framework when they're not actually "inline" but moved to a separate test file. It feels like we get the worst of both approaches by doing that.

Reply:
> I feel like the whole "inline tests as a testing framework" thing doesn't work very well with the current tooling or at least when having several opam packages in the same repo

Yes, there are some thing that I regret about with the current designs. We should have really added an inline_tests stanza.

> I have to admit that I'm not sure what the benefit of inline tests is compared to a regular testing framework when they're not actually "inline" but moved to a separate test file. It feels like we get the worst of both approaches by doing that.

ppx_expect has a great workflow with promotion. I believe that's the only reason.

Reply:
Yeah that's a good point, it is very pleasant to work with compared to other existing testing framework.

> Yes, there are some thing that I regret about with the current designs. We should have really added an inline_tests stanza.

It's not too late to add one!